syntax = "proto3";
package player;
import "common/common.proto";
service Player {
  rpc Init (InitRequest) returns (InitResponse) {}
  rpc Move (MoveRequest) returns (MoveResponse) {}
  rpc Region (RegionRequest) returns (RegionResponse) {}
  rpc MassIncrement (MassIncrementRequest) returns (MassIncrementResponse) {}
}

message InitRequest {
}

message InitResponse {
  string ip = 1;
  double x = 2;
  double y = 3;
  int32 mass = 4;
  int32 ver = 5;
}
message MoveRequest {
  string ip = 1;
  double x = 2;
  double y = 3;
  int32 ver = 4;
}
message MoveResponse {
  double x = 1;
  double y = 2;
  bool alive = 3;
  int32 mass = 4;
}
message RegionRequest {
  double x = 1;
  double y = 2;
}
message RegionResponse {
  repeated common.Blob blobs = 1;
  repeated common.Food foods = 2;
}
message MassIncrementRequest {
  int32 massIncrease = 1;
}
message MassIncrementResponse {
  
}